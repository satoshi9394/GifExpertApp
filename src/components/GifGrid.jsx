import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {

  const { data: images , loading} = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn animate__delay-1">{category}</h3>
      {
        loading && <p className="animate__animated animate__flash animate__delay-1">Loading...</p>
      }
      <div className="card-grid">
        {
          images.map( img => (
            <GifGridItem 
              key={img.id}
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid
