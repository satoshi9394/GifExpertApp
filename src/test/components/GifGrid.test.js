import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); 

describe('Pruebas en <GifGrid />', () => {
  const category = 'Goku'

  test('debe mostrar el componente correctamente ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow( <GifGrid category={category} /> );
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes con nuestro useFetchGifs ', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/img.jpg',
      title: 'goku'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })
    
    const wrapper = shallow( <GifGrid category={category} /> );
    expect( wrapper ).toMatchSnapshot();
  });

})