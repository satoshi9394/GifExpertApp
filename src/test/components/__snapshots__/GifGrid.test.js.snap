// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes con nuestro useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="goku"
      url="https://localhost/img.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash animate__delay-1"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
